floc.bs

```diff
@@ -1 +1,169 @@
+
+  This specification provides an API to enable ad-targeting based on the people’s general browsing interest, without exposing the exact browsing history.
+
+  <div class="example">
+    Creating an ad based on the interest cohort:
+
+    <pre class="lang-js">
+      const cohort = await document.interestCohort();
+      const url = new URL("https://ads.example/getCreative");
+      url.searchParams.append("cohort_id", cohort.id);
+      url.searchParams.append("cohort_version", cohort.version);
+      const creative = await fetch(url);
+    </pre>
+  </div>
+</section>
+
+<section>
+  <h2 id="interest-cohort-section">Interest cohort</h2>
+
+  The <dfn>interest cohort</dfn> is a user's assigned interest group under a particular <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a>. An interest cohort comprises an [=interest cohort id=] and an [=interest cohort version=].
+
+  The <dfn>interest cohort id</dfn> represents the interest group that the user is assigned to by the <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a>. The total number of groups should not exceed 2^32, and each group can mapped to a 32 bit integer. The interest cohort id can be invalid, which means no group is assigned.
+
+  The <dfn>string representation of the interest cohort id</dfn> is the string representation of the mapped integer of the [=interest cohort id=] in decimal (e.g. “17319”). If the [=interest cohort id=] is invalid, the string representation will be an empty string.
+
+  The <dfn>interest cohort version</dfn> identifies the <a href="#cohort-assignment-algorithm">algorithm</a> used to compute the [=interest cohort id=].
+
+  The <dfn>string representation of the interest cohort version</dfn> is browser defined. It’s recommended to make the browser vendor name as part of the version (e.g. “chrome.2.1”, “v21/mozilla”), so that when exposed to the Web, there won't be naming collisions across browser vendors.
+
+  <h3 id="interest-cohort-dictionary">Dictionary {{InterestCohort}}</h3>
+  The {{InterestCohort}} dictionary is used to contain the [=string representation of the interest cohort id=] and the [=string representation of the interest cohort id=].
+
+  <pre class="idl">
+  dictionary InterestCohort {
+    DOMString id;
+    DOMString version;
+  };
+  </pre>
+</section>
+
+<section>
+  <h2 id="the-api">The API</h2>
+
+  The interest cohort API lives under the {{document}} object since the access permission is tied to the document scope.
+
+  <pre class="idl">
+    partial interface Document {
+        Promise&lt;DOMString&gt; interestCohort();
+    };
+  </pre>
+
+  The <dfn for="Document" method>interestCohort()</dfn> method steps are:
+  1. Let |p| be a new {{Promise}}.
+  2. If the document is not allowed to use the "<code><a href="#interest-cohort-policy-controlled-feature">interest-cohort</a></code>" feature, reject and then return |p|.
```

Comment:
```suggestion
  2. If [=this=] is not [=allowed to use=] the "<code><a href="#interest-cohort-policy-controlled-feature">interest-cohort</a></code>" feature, [=reject=] and then return |p|.
```

But, more important issue: what are you rejecting |p| with? Maybe `[=reject=] |p| with a "{{NotAllowedError}}" {{DOMException}} and return |p|`?

Same for all other rejections. If the plan is to reject them all with the same type of error, then I'd suggest a big if statement, e.g.

```
2. If any of the following is true:
    - ...
    - ...
    - ...
  then [=reject=] |p| with ...
```

Reply:
Done.